### ISSUE TITLE
Make it tree-shakable

### ISSUE THREAD

**Issue by 5d-jh** [photopea/UPNG.js#81]
I'm using this library with bundled by Webpack.

In my case, I only use `decode` function. However since this library is wrapped by an object, bundled file includes `pako` which seems not used when decoding.

This results unnecessary large bundle size.

**Comment by photopea** [comment]
yes, pako.js is needed only for encoding, not for decoding. You can manually delete the encoding parts from the UPNG.js file.

**Comment by 5d-jh** [comment]
Yes, for now I'm using this library with encoding parts deleted.

But for everyone(including me) that want to use tree-shaked version, removing implementatinos that are not familiar with doesn't seems right.

Since this library doesn't have a lot of functions, it can be easily done. If you are okay with this, may I take over this issue?

**Comment by photopea** [comment]
What are your plans? Do you want to make two files? There are certain parts needed both by an encoder and by a decoder.

**Comment by 5d-jh** [comment]
No I'm not separating into multiple files. Just going to use `export` clause to each public function so let bundlers know which functions are actually used.

This is example of tree-shakable version that I was talking of:

```
import pako from 'pako';

export function encode() {
   ...
  someInternal()
}

export function decode() {
  pako.dosomething();
  someInternal()
  ...
}

function someInternal() {
 ...
}
```

**Comment by 5d-jh** [comment]
My plan is publish tree-shakable version as v3, and keep current version stay in v2 to prevent you from forcing adopt new toolchains(webpack, npm, etc...).

**Comment by lvcabral** [comment]
I also think to use `pako` makes this package bigger than it needs, can't you use something like:
https://www.npmjs.com/package/fflate

I made a fork and changed the module to `fflate`
https://github.com/lvcabral/UPNG.js

**Comment by danielbarion** [comment]
Hey guys! I found this project while testing photopea (also, great work on it @photopea!)

About the bundle size, if UPNG is tree-shakeable this will save almost 75% of the bundle size:

@photopea please take a look at this:

https://bundlephobia.com/package/upng-js@2.1.0

![image](https://github.com/photopea/UPNG.js/assets/9615850/9cd98e28-13d1-4031-8864-6c9c7634fb57)

**Comment by photopea** [comment]
@danielbarion I am so confused. The UPNG.js library is 46.6 kB now, and it is 14.5 kB GZipped. If you want it to be small, do not use that tool that you are referring to.
